R/diffnet-class.r

```diff
@@ -629,21 +672,40 @@ new_diffnet <- function(graph, toa, t0=min(toa, na.rm = TRUE), t1=max(toa, na.rm
   meta$multiple   <- multiple
   meta$name       <- ifelse(!length(name), "", ifelse(is.na(name), "",
                                                       as.character(name)))
-  meta$behavior   <- ifelse(!length(behavior), "", ifelse(is.na(behavior), "",
-                                                          as.character(behavior)))
   meta$version    <- utils::packageVersion("netdiffuseR")
 
   # Removing dimnames
   graph                  <- Map(function(x) Matrix::unname(x), x=graph)
   dimnames(toa)          <- NULL
-  dimnames(mat$adopt)    <- NULL
-  dimnames(mat$cumadopt) <- NULL
+
+  if (num_of_behaviors==1) {
+    meta$behavior   <- ifelse(!length(behavior), "", ifelse(is.na(behavior), "",
```

Comment:
The `behavior` entry should be a vector. What happens if the behavior has a name with a comma in it? Many parts of the code will break (b/c use use strplit(,)). It should be pasted when you print it out only.